libcylinder/tests/mod.rs

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2018-2020 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+use std::thread;
+
+use cylinder::{Context, PrivateKey, Signature};
+
+type TestResult = Result<(), Box<dyn std::error::Error>>;
+
+fn test_multithreaded_signing<C: Context + 'static>(context: C) -> TestResult {
+    let private_key = PrivateKey::new_from_hex(
+        "2f1e7b7a130d7ba9da0068b3bb0ba1d79e7e77110302c9f746c3c2a63fe40088",
```

Comment:
This is secp256k1-specific, so should probably be passed into the fn.

Reply:
Fixed